src/main/java/com/fasterxml/jackson/core/Base64Variant.java

```diff
@@ -166,6 +182,47 @@ public Base64Variant(Base64Variant base, String name, boolean usesPadding, char
         _usesPadding = usesPadding;
         _paddingChar = paddingChar;
         _maxLineLength = maxLineLength;
+        this._paddingReadBehaviour = paddingReadBehaviour;
+    }
+
+    private Base64Variant(Base64Variant base, PaddingReadBehaviour paddingReadBehaviour) {
+        this(base, base._name, base._usesPadding, base._paddingChar, paddingReadBehaviour, base._maxLineLength);
+    }
+
+    /**
+     * @since 2.12
+     */
+    public Base64Variant withPaddingAllowed() {
+        return new Base64Variant(this, PaddingReadBehaviour.PADDING_ALLOWED);
+    }
+
+    /**
+     * @since 2.12
+     */
+    public Base64Variant withPaddingRequired() {
+        return new Base64Variant(this, PaddingReadBehaviour.PADDING_REQUIRED);
+    }
+
+    /**
+     * @since 2.12
+     */
+    public Base64Variant withPaddingForbidden() {
+        return new Base64Variant(this, PaddingReadBehaviour.PADDING_FORBIDDEN);
+    }
+
+    /**
+     * @since 2.12
+     */
+    public Base64Variant withWritePadding(boolean writePadding) {
+        return new Base64Variant(this, this._name, writePadding, this._paddingChar, this._maxLineLength);
+
+    }
+
+    private enum PaddingReadBehaviour {
```

Comment:
Since exposed by public accessor, needs to be public type (I can change this post-merge).

Reply:
Changed and added comments and `@since`